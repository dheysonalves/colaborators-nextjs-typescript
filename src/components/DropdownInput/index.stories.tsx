import React, { useState } from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import DropdownInput from '.'

export default {
	title: 'Inputs/DropdownInput',
	component: DropdownInput,
	args: {
		label: 'Departamento',
		data: ['Option 1', 'Option 2', 'Option 3'],
	},
} as ComponentMeta<typeof DropdownInput>

export const Default: ComponentStory<typeof DropdownInput> = (args) => (
	<DropdownInput {...args} />
)

export const WithoutLabel: ComponentStory<typeof DropdownInput> = () => {
	const [value, setValue] = useState<string | number>('')

	return (
		<DropdownInput
			hasLabel={false}
			label="Departamento"
			data={['Option 1', 'Option 2', 'Option 3']}
			value={value as string}
			setValue={setValue}
		/>
	)
}
