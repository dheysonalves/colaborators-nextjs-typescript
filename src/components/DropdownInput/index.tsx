import React, { useState } from 'react'
import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/io'
import {
	AbsoluteIcon,
	Container,
	Select,
	SelectContainer,
	Option,
} from './styles'

interface DropdownInputProps {
	label?: string
	data: string[] | number[]
	hasLabel?: boolean
	value: string
	setValue: (value: string | number) => void
}

const DropdownInput = ({
	label,
	data,
	hasLabel = true,
	value,
	setValue,
}: DropdownInputProps) => {
	const [dropdownIconState, updateDropdownIconState] = useState(false)
	const ICON_SIZE = 14

	return (
		<Container>
			{hasLabel && <label htmlFor="standart-select">{label}</label>}
			<SelectContainer>
				<Select
					id="standart-select"
					onClick={() => updateDropdownIconState(!dropdownIconState)}
					value={value}
					onChange={(e) => setValue(e.target.value)}
					aria-label="A select for the user"
				>
					{data.map((item, index) => {
						return (
							<Option
								value={item.toString().toLocaleLowerCase()}
								key={index.toString()}
							>
								{item}
							</Option>
						)
					})}
				</Select>
				<AbsoluteIcon>
					{dropdownIconState ? (
						<IoIosArrowUp size={ICON_SIZE} color="#587169" />
					) : (
						<IoIosArrowDown size={ICON_SIZE} color="#587169" />
					)}
				</AbsoluteIcon>
			</SelectContainer>
		</Container>
	)
}

export default DropdownInput
