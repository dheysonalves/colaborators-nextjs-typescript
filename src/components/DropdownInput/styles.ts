import styled from '@emotion/styled'

export const Container = styled.div`
	position: relative;
	background-color: ${(props) => props.theme.light.neutral.neutral_2};
	border: 2px solid ${(props) => props.theme.light.neutral.neutral_1};
	border-radius: 8px;

	label {
		position: absolute;
		top: -10px;
		left: 10px;
		background-color: ${(props) => props.theme.light.neutral.neutral_2};
		color: ${(props) => props.theme.light.neutral.neutral_3};
		padding: 0 1em;
		font-size: 14px;
	}
`

export const SelectContainer = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
`

export const Select = styled.select`
	width: 100%;
	border-radius: 8px;
	padding: 1.5em 1em;
	font-size: 16px;
	font-weight: 500;
	cursor: pointer;
	line-height: 1.1;
	color: ${(props) => props.theme.light.neutral.neutral_5};
	background-color: ${(props) => props.theme.light.neutral.neutral_2};
`

export const Option = styled.option`
	font-size: 16px;
	font-weight: 500;
	color: ${(props) => props.theme.light.neutral.neutral_5};
`

export const AbsoluteIcon = styled.div`
	position: absolute;
	right: 0;
	margin-right: 2em;
`
