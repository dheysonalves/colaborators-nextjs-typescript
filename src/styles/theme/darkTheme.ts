export const theme = {
	dark: {
		brand: {
			blue: '#034AFD',
		},
		neutral: {
			background: '#F8FAF9',
			light: '#fff',
			black: '#34423D',
			neutral_2: '#F5FAF8',
			neutral_3: '#A3B8B0',
			neutral_5: '#587169',
		},
		primary: {
			default: '#22E0A1',
			green: '#B5F1DD',
		},
		feedback: {
			success: '#B5F1DD',
		},
		tertiary: {
			disabled: '#EAEFED',
		},
	},
}
