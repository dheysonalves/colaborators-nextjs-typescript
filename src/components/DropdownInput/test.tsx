import React from 'react'

import { render, screen } from '@testing-library/react'

import { ThemeProvider } from '@emotion/react'

import DropdownInput from '.'
import { theme } from '../../styles/theme/theme'

const setup = () => {
	const updateValue = (value: string) => {
		return value
	}

	const utils = render(
		<ThemeProvider theme={theme}>
			<DropdownInput
				label="Departamento"
				data={['1', '2', '3']}
				value="1"
				setValue={() => updateValue('2')}
			/>
			,
		</ThemeProvider>,
	)
	return {
		...utils,
	}
}

describe('<DropdownInput />', () => {
	it('should render the heading', () => {
		const { container } = setup()

		expect(screen.getByText('Departamento')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
