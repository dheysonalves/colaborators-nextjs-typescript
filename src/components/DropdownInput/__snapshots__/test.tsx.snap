// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownInput /> should render the heading 1`] = `
<div
  class="css-185xvve"
>
  <label
    for="standart-select"
  >
    Departamento
  </label>
  <div
    class="css-17yg843"
  >
    <select
      aria-label="A select for the user"
      class="css-5puhau"
      id="standart-select"
    >
      <option
        class="css-salr3s"
        value="1"
      >
        1
      </option>
      <option
        class="css-salr3s"
        value="2"
      >
        2
      </option>
      <option
        class="css-salr3s"
        value="3"
      >
        3
      </option>
    </select>
    <div
      class="css-hd2x4e"
    >
      <svg
        color="#587169"
        fill="currentColor"
        height="14"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(88, 113, 105);"
        viewBox="0 0 512 512"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
        />
      </svg>
    </div>
  </div>
</div>
`
